product_multi_lot_sequence/models/stock_production_lot.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 ForgeFlow S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductionLot(models.Model):
+    _inherit = "stock.production.lot"
+
+    product_lot_sequence_id = fields.Many2one(
+        "product.lot.sequence",
+        domain="['|',('product_id','=', product_id),"
+        "'&',('product_id','=', False),('product_tmpl_id','=', product_tmpl_id)]",
+    )
+    available_product_lot_sequence_ids = fields.Many2many(
+        "product.lot.sequence", compute="_compute_available_product_lot_sequence_ids"
+    )
+    available_product_lot_sequence_count = fields.Integer(
+        compute="_compute_available_product_lot_sequence_ids"
+    )
+    product_tmpl_id = fields.Many2one(
+        "product.template", related="product_id.product_tmpl_id"
+    )
+
+    @api.depends("product_id")
+    def _compute_available_product_lot_sequence_ids(self):
+        for rec in self:
+            if rec.product_id:
+                pls_ids = rec.product_id.product_tmpl_id.product_lot_sequence_ids
+                available = pls_ids.filtered(lambda x: x.product_id == self.product_id)
+                if not available:
+                    available = pls_ids.filtered(
+                        lambda x: x.product_id.product_tmpl_id == self.product_tmpl_id
+                    )
+                rec.available_product_lot_sequence_ids = available
+                rec.available_product_lot_sequence_count = len(available)
+            else:
```

Comment:
product_id is required on lot. Is this necessary ?

Reply:
I was trying to keep product lot sequence field hidden if there is only one option. What do you suggest?